SWENT-GRP5/SOLVIT
# Consistent theme colors

# Purpose:
This pull request introduces the use of Material Theme's color scheme throughout the app for both Light and Dark modes.

- **What issue does it solve?** This PR resolves the lack of consistency in colors and the overall UI throughout the app
- **What functionality does it add?** With the introduction of the Material Theme, any developer can now easily code with the correct colors while checking compatibility with Dark mode

### Related Issues:
- #261 

# Changes Made:
- Set Colors.kt to establish a theme with respect to our main colors (Green and Blue)
- Set Theme.kt to use these colors and adapt whether the app uses Light or Dark mode
- Set the Provider Profile screen to match better the Figma colors
- [Fixed signUp_emailAndPasswordInput test](https://github.com/SWENT-GRP5/SOLVIT/pull/287/commits/e3138b81703610d1dce468cbdd89cf22f9d63be3)
- Improved the Provider Info tab selected indicator
- Introduced tests for `Color.kt` and `Theme.kt`
- Improved UI of `SelectedProvider.kt` to fit the Figma better by adding shadows and modifying padding

# How to Test:
1. Run the app on phone or emulator
2. Navigate through all screens.
3. Check that UI is clear and has enough contrast in both Light and Dark conditions

<details>
<summary>Visual Helpers</summary>

##### Light Theme
1. Seeker
!(https://github.com/user-attachments/assets/1d60c7fd-b9bf-48e9-98af-8042d4f96c6a)

2. Provider
!(https://github.com/user-attachments/assets/196a3bfa-5c81-4866-a14e-100d28c6eb8d)


##### Dark Theme
1. Seeker
!(https://github.com/user-attachments/assets/1898231f-89db-4b23-9583-507d99ca418f)

2. Provider
!(https://github.com/user-attachments/assets/8367c8e0-1537-482a-8d00-40ed51561401)

</details>

# Comments:
- `ProviderCalendarScreen.kt` and `SeekerProfile.kt` are addressed in another PR (both are Theme resilient but not optimally implemented to not lose too much time and make this PR as clear as possible 

---

# Checklist:
